src/compiler/checker.ts

```diff
@@ -38217,8 +38217,9 @@ namespace ts {
             checkSourceElement(node.elseStatement);
         }
 
-        function checkTestingKnownTruthyCallableOrAwaitableType(condExpr: Expression, body?: Statement | Expression) {
+        function checkTestingKnownTruthyCallableOrAwaitableType(condExpr: Expression, condType: Type, body?: Statement | Expression) {
             if (!strictNullChecks) return;
+            if (!(getTypeFacts(condType) & TypeFacts.Truthy)) return;

```

Comment:
Shouldn't this check only be done if `location` is ultimately `condExpr`?

Reply:
I think you are correct; as this check is indeed also done within `helper` based on the resolved type. Removing this.